llama-index-integrations/tools/llama-index-tools-agentql/llama_index/tools/agentql/base.py

```diff
@@ -0,0 +1,227 @@
+        response_mode: Optional[str] = DEFAULT_RESPONSE_MODE,
+        stealth_mode: Optional[bool] = False,
+        is_scroll_to_bottom_enabled: Optional[bool] = False,
+        is_screenshot_enabled: Optional[bool] = False,
+    ) -> None:
+        """
+        Initialize AgentQLToolSpec.
+
+        Args:
+            async_browser: A browser instance to use for automation.
+            extract_data_timeout: Timeout value in seconds for the connection with Extract Data service.
+            extract_elements_timeout: Timeout value in seconds for the connection with Extract Elements service.
+            wait_for_page_load: Wait time in seconds for page load completion.
+            wait_for_network_idle: Whether to wait for network reaching full idle state before querying the page.
+            include_hidden_data: Whether to include hidden elements on the page for extract data.
+            include_hidden_elements: Whether to include hidden elements on the page for extract elements.
+            response_mode: The mode of the query. It can be either 'standard' or 'fast'.
+            stealth_mode: Whether to run the browser in stealth mode. This is useful for avoiding detection by anti-bot services (extract_web_data only).
+            is_scroll_to_bottom_enabled: Enable scrolling to bottom of the page before extracting data (extract_web_data only).
+            is_screenshot_enabled: Whether to take a screenshot of the page before extracting data (extract_web_data only).
+        """
+        self.async_browser = async_browser
+
+        self.extract_data_timeout = extract_data_timeout
+        self.extract_elements_timeout = extract_elements_timeout
+
+        self.wait_for_page_load = wait_for_page_load
+        self.wait_for_network_idle = wait_for_network_idle
+
+        self.include_hidden_data = include_hidden_data
+        self.include_hidden_elements = include_hidden_elements
+
+        self.response_mode = response_mode
+        self.stealth_mode = stealth_mode
+
+        self.is_scroll_to_bottom_enabled = is_scroll_to_bottom_enabled
+        self.is_screenshot_enabled = is_screenshot_enabled
+
+    @classmethod
+    def from_async_browser(cls, async_browser: AsyncBrowser) -> "AgentQLToolSpec":
+        """
+        Initialize AgentQLToolSpec from an async browser instance.
+        """
+        return cls(async_browser=async_browser)
+
+    @staticmethod
+    async def create_async_playwright_browser(
+        headless: bool = True, args: Optional[List[str]] = None
+    ) -> AsyncBrowser:
+        """
+        Create an async playwright browser.
+
+        Args:
+            headless: Whether to run the browser in headless mode. Defaults to True.
+            args: arguments to pass to browser.chromium.launch
+
+        Returns:
+            AsyncBrowser: The playwright browser.
+        """
+        from playwright.async_api import async_playwright
+
+        browser = await async_playwright().start()
```

Comment:
it's not broswer, it's playwright
```suggestion
        playwright = await async_playwright().start()
```